internal/mode/static/telemetry/collector.go

```diff
@@ -275,3 +263,56 @@ func CollectClusterID(ctx context.Context, k8sClient client.Reader) (string, err
 	}
 	return string(kubeNamespace.GetUID()), nil
 }
+
+// CollectNodeCount returns the number of nodes in the cluster.
+func CollectNodeCount(ctx context.Context, k8sClient client.Reader) (int, error) {
+	var nodes v1.NodeList
+	if err := k8sClient.List(ctx, &nodes); err != nil {
+		return 0, fmt.Errorf("failed to get NodeList: %w", err)
+	}
+
+	return len(nodes.Items), nil
+}
+
+type clusterInformation struct {
+	Platform  string
+	Version   string
+	ClusterID string
+}
+
+func collectClusterInformation(ctx context.Context, k8sClient client.Reader) (clusterInformation, error) {
+	var clusterInfo clusterInformation
+	var err error
+
+	var nodes v1.NodeList
+	if err = k8sClient.List(ctx, &nodes); err != nil {
+		return clusterInformation{}, fmt.Errorf("failed to get NodeList: %w", err)
+	}
+	if len(nodes.Items) == 0 {
+		return clusterInformation{}, errors.New("failed to collect cluster information: NodeList length is zero")
+	}
+
+	var clusterID string
+	if clusterID, err = CollectClusterID(ctx, k8sClient); err != nil {
+		return clusterInformation{}, fmt.Errorf("failed to collect cluster information: %w", err)
+	}
+	clusterInfo.ClusterID = clusterID
+
+	node := nodes.Items[0]
```

Comment:
I'd move the block
```

	var nodes v1.NodeList
	if err = k8sClient.List(ctx, &nodes); err != nil {
		return clusterInformation{}, fmt.Errorf("failed to get NodeList: %w", err)
	}
	if len(nodes.Items) == 0 {
		return clusterInformation{}, errors.New("failed to collect cluster information: NodeList length is zero")
	}
```
next too
```
	node := nodes.Items[0]
```
so that related lines are close together